Microsoft.Azure.CosmosRepository/src/Extensions/ExpressionExtensions.cs

```diff
@@ -41,5 +42,23 @@ internal static class ExpressionExtensions
         internal static Expression<Func<T, bool>> Or<T>(
             this Expression<Func<T, bool>> first,
             Expression<Func<T, bool>> second) => first.Compose(second, Expression.Or);
+
+        internal static PropertyInfo GetPropertyInfo<TSource, TProperty>(this Expression<Func<TSource, TProperty>> propertyLambda)
+        {
+            Type type = typeof(TSource);
+
+            MemberExpression member = propertyLambda.Body as MemberExpression;
+            if (member == null)
+                throw new ArgumentException($"Expression '{propertyLambda}' refers to a method, not a property.");
+
+            PropertyInfo propInfo = member.Member as PropertyInfo;
+            if (propInfo == null)
+                throw new ArgumentException($"Expression '{propertyLambda}' refers to a field, not a property.");

```

Comment:
```suggestion
            if (propertyLambda.Body is not MemberExpression member)
                throw new ArgumentException($"Expression '{propertyLambda}' refers to a method, not a property.");

            if (member.Member is not PropertyInfo propInfo)
                throw new ArgumentException($"Expression '{propertyLambda}' refers to a field, not a property.");
```

I believe we could use the `is not` operator instead of `as` and `null` check. But you'd have to verify.